awscrt/mqtt.py

```diff
@@ -110,12 +110,14 @@ class Will:
         qos (QoS): QoS used when publishing the Will message.
         payload (bytes): Content of Will message.
         retain (bool): Whether the Will message is to be retained when it is published.
+            This is not supported by AWS IoT. See https://docs.aws.amazon.com/iot/latest/developerguide/mqtt.html
```

Comment:
Thanks, this is good to call out, but let's keep it ambiguous in case this feature is added in the future and we forget to update all the docs everywhere. I know the AWS IoT team is always weighing the possibility of supporting more MQTT features.

```Retain is not supported by AWS IoT as of 2020, see https://docs.aws.amazon.com/iot/latest/developerguide/mqtt.html for current status.```